### DISCUSSION TITLE
Urgent help - How do I override default behaviour of appautoscaling to treat missing data as good.

### DISCUSSION THREAD

**Discussion by PartTimer1996** [aws/aws-cdk#21331] [Q&A]
I am deploying a complex Stack that creates a Fargate service than assigns app auto scaling on CPUutilisation. However, this auto scaling automatically creates an alarm for me, which is fine, but this alarm treats missing data as missing leading to issues with the alarm flagging when there is simply no data to access. I really need to override this and allow for the default behaviour of the Alarm to allow for missing data to be good. Ideally I could just override this with a simple: treatMissingData: notBreaching. 

Code snippet can be seen below. Any help would be greatly appreciated! 

        export class AppAutoScalingDeploy extends Construct {
	        constructor(scope: Construct, id: string, props: AppAutoScalingDeployProps) {
		super(scope, id);
		const runningAppMetric = new cloudwatch.Metric({
			namespace: `AWS/ECS`,
			metricName: 'CPUUtilization',
			period: Duration.minutes(1),
			statistic: 'Average',
			dimensionsMap: { ClusterName: props.clusterName, ServiceName: props.serviceName },
		});
		const scalingTarget = new autoscaling.ScalableTarget(this, 'AppScalingTarget', {
			maxCapacity: 10,
			minCapacity: 0,
			resourceId: 'service/${props.clusterName}/${props.environment}_${props.appName}_service',
			scalableDimension: 'ecs:service:DesiredCount',
			serviceNamespace: autoscaling.ServiceNamespace.ECS,
		});

		new autoscaling.StepScalingPolicy(this, 'AppScalingPolicy', {
			adjustmentType: autoscaling.AdjustmentType.EXACT_CAPACITY,
			scalingTarget: scalingTarget,
			metric: runningAppMetric,
			evaluationPeriods: 15,
			datapointsToAlarm: 15,
			scalingSteps: [
				{ lower: 0, upper: 1, change: 0 },
				{
					lower: 1,
					upper: 2,
					change: 0,
				},
			],
		});

		new autoscaling.StepScalingPolicy(this, 'AppHighUsageAlarm', {
			adjustmentType: autoscaling.AdjustmentType.EXACT_CAPACITY,
			scalingTarget: scalingTarget,
			metric: runningAppMetric,
			cooldown: Duration.minutes(15),
			evaluationPeriods: 1,
			datapointsToAlarm: 1,
			scalingSteps: [
				{ lower: 2, upper: 10, change: 2 },
				{
					lower: 10,
					upper: 50,
					change: 4,
				},
				{
					lower: 50,
					change: 5,
				},
			],
		});
	}
}

**Comment by enigma87** [comment]
I'm facing the same problem, leaving a comment here to follow.
